### ISSUE TITLE
Window support soon or is it not possible

### ISSUE THREAD

**Issue by Xtaker-of-lifeX** [axi0mX/ipwndfu#48]
Would it be possible to get window support or is there a way to boot linux on a windows computer

**Comment by 2Eggsss** [comment]
> Would it be possible to get window support or is there a way to boot linux on a windows computer

You could use a live cd (Ubuntu live CD) or just dual boot into linux by installing it in another partition.

**Comment by Xtaker-of-lifeX** [comment]
My computer runs like dog shit so dual boot wont be possible

**Comment by 2Eggsss** [comment]
> My computer runs like dog shit so dual boot wont be possible

You "might" be able to get away with using a live CD. If that doesn't work then there is no other way.

**Comment by Hendrikto** [comment]
I have not tested this, but it might work under WSL2, as it actually runs a Linux kernel instead of emulating it.

**Comment by 2Eggsss** [comment]
> I have not tested this, but it might work under WSL2, as it actually runs a Linux kernel instead of emulating it.

It still runs in a VM so it may not work, I have not tested it myself either but just stating facts.

**Comment by Xtaker-of-lifeX** [comment]
So is there a way to just boot linux on my computer and not use windows 10

**Comment by mrkushy** [comment]
Its python just download it on your windows

**Comment by dznas** [comment]
> > I have not tested this, but it might work under WSL2, as it actually runs a Linux kernel instead of emulating it.
> 
> It still runs in a VM so it may not work, I have not tested it myself either but just stating facts.

Will the ipad6 running 12.4.1 be downgraded?

**Comment by dznas** [comment]
> 它的python只需将其下载到Windows

Can you write a tutorial? For non-programmers, only windows, thank you

> Its python just download it on your windows

Can you write a tutorial? For non-programmers, only windows, thank you

**Comment by jmichaelbarker** [comment]
> Can you write a tutorial? For non-programmers, only windows, thank you

There's not really much that can be done with this as a non-programmer at the moment. And besides, you almost certainly would require more than just python, like libusb (I haven't got a clue what its support looks like on windows) and potentially more. It would definitely be simpler to use a linux livecd.

**Comment by Hendrikto** [comment]
A tutorial in using LiveCDs can be found [here](https://help.ubuntu.com/community/LiveCD). It contains links to other articles about accomplishing the same with a USB stick instead.

This may be scary/difficult for some people, as it requires creating a bootable medium, in addition to to changing your BIOS bootorder and/or entering the boot menu. If you are uncomfortable doing this, this might be an indicator that you are not the target audience for this.

**Comment by 2Eggsss** [comment]
> A tutorial on using LiveCDs can be found [here](https://help.ubuntu.com/community/LiveCD). It contains links to other articles about accomplishing the same with a USB stick instead.
> 
> This may be scary/difficult for some people, as it requires creating a bootable medium, in addition to changing your BIOS bootorder and/or entering the boot menu (as discussed [here](https://help.ubuntu.com/community/BootFromCD)). If you are uncomfortable doing this, this might be an indicator that you are not the target audience for this.

Best thing to do if you don't wanna go the LiveCD way is to just WAIT. This is a program made for developers to implement their own methodologies into practice and is not really suited for consumers to do anything with (you can't do much anyways besides debugging etc and dumping the SecureROM). THIS IS NOT A JAILBREAK!!!

**Comment by zombodotcom** [comment]
Could probably use a VM as well.

**Comment by 2Eggsss** [comment]
> Could probably use a VM as well.

Check, it says that it will not work in a VM.

**Comment by Ronsor** [comment]
I tried getting it running on windows but there's an error: `ValueError: the device has no langid`

**Comment by dznas** [comment]
> > Can you write a tutorial? For non-programmers, only windows, thank you
> 
> There's not really much that can be done with this as a non-programmer at the moment. And besides, you almost certainly would require more than just python, like libusb (I haven't got a clue what its support looks like on windows) and potentially more. It would definitely be simpler to use a linux livecd.

感谢你的回复
thank you for your reply

**Comment by jmichaelbarker** [comment]
> I tried getting it running on windows but there's an error: `ValueError: the device has no langid`

I'm not sure how the Windows permission model applies to USB/HID devices (or whatever Windows decides a DFU iDevice is) but on both Linux distros I've tested, libusb is only able to communicate with the device as root; so, if you're not already, try executing the script with higher privileges (from an Admin console, for instance).

**Comment by zombodotcom** [comment]
I can look into it for Windows, but honestly just get etcher.io and make a live Linux ( Ubuntu or some distro) live boot. Google search. Just boot your system from the live boot

**Comment by DanHolli123** [comment]
@2Eggsss it is sandboxed, which is different from a vm because it does run natively, just blocked from some things

**Comment by 2Eggsss** [comment]
> @2Eggsss it is sandboxed, which is different from a vm because it does run natively, just blocked from some things

Please test and reply if it works with WSL2, I will inform+apply as required.

**Comment by Millss23** [comment]
i have a 64 bit computer will it work it’s pretty good and how do i do it i’m big noob

**Comment by Ronsor** [comment]
> > I tried getting it running on windows but there's an error: `ValueError: the device has no langid`
> 
> I'm not sure how the Windows permission model applies to USB/HID devices (or whatever Windows decides a DFU iDevice is) but on both Linux distros I've tested, libusb is only able to communicate with the device as root; so, if you're not already, try executing the script with higher privileges (from an Admin console, for instance).

I tried that actually.

**Comment by drewwalton19216801** [comment]
@Miles23, if you are that unfamiliar with this, this may not be for you. This is just the exploit. It *does not* jailbreak your device.

**Comment by EWouters** [comment]
Geohot got it to work on Windows, check the [stream](https://www.youtube.com/watch?v=0f21HU2Lr2o) and his fork of the [repo](https://github.com/geohot/ipwndfu) (he ported it to Python 3 as well).

Edit: actually he didn’t get it to work and didn’t really port it to Python 3 😑

**Comment by Millss23** [comment]
> @Millss23, if you are that unfamiliar with this, this may not be for you. This is just the exploit. It _does not_ jailbreak your device.

ok tnx will do

**Comment by EWouters** [comment]
Please check #67 and close this issue.

**Comment by 0n1cOn3** [comment]
Check out our Pionieer Geohot (George Hotz) his Github Repo. Also check the stream from twich, which he has successfully dumped the BootROM from his iPhone 7 on Windows.

**Comment by babarqb** [comment]
> Check out our Pionieer Geohot (George Hotz) his Github Repo. Also check the stream from twich, which he has successfully dumped the BootROM from his iPhone 7 on Windows.

No Geohot not able to pwnd his iPhone 7 on the window he tried very hard but in the end, he used a mac and got success.

**Comment by xp2-882030kgz010602** [comment]
Would it work to compile with Cygwin?

**Comment by drewwalton19216801** [comment]
@xp2-882030kgz010602 No, Cygwin would not help. This is Python scripting, with ARM shellcode that needs to be compiled with a specific ARM assembler.

**Comment by Ronsor** [comment]
> @xp2-882030kgz010602 No, Cygwin would not help. This is Python scripting, with ARM shellcode that needs to be compiled with a specific ARM assembler.

That doesn't even have anything to do with it. The software itself runs fine on Windows, but libusb doesn't work right.

**Comment by adelphiamalloc** [comment]
any particular preference of linux?? i've tried a few without much success in the past.. i love my kali but i don't think it's applicable here.. lmk adelphiamalloc @ gmail.com thanx
